torch/csrc/generic/StorageMethods.cpp

```diff
@@ -190,13 +190,33 @@ PyObject * THPStorage_(newWithFile)(PyObject *_unused, PyObject *file)
   int fd = PyObject_AsFileDescriptor(file);
   THPUtils_assert(fd != -1, "_new_with_file couldn't retrieve a file "
       "descriptor from given object");
-  THStoragePtr storage = THPStorage_(readFileRaw)(fd);
+  THStoragePtr storage = THPStorage_(readFileRaw)(fd, nullptr);
   PyObject *result = THPStorage_(New)(storage);
   storage.release();
   return result;
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject *THPStorage_(setFromFile)(THPStorage *self, PyObject *args)
+{
+  HANDLE_TH_ERRORS
+  PyObject *file = PyTuple_GET_ITEM(args, 0);
+  int fd = PyObject_AsFileDescriptor(file);
+
+  PyObject *offset = PyTuple_GET_ITEM(args, 1);
```

Comment:
Are you sure offsets will always by `PyLong`s? I think it's better to use `THPUtils_getLong` (and it will throw an exception if it fails).